### DISCUSSION TITLE
How can I centralize figure caption in Quarto with docx output?

### DISCUSSION THREAD

**Discussion by mjdzr** [quarto-dev/quarto-cli#11513] [Q&A] [answered]
### Description

I have created a new template document, which is essentially a totally new document with the "Image Caption" style changed to Arial 9 and making sure it's center aligned:

![image](https://github.com/user-attachments/assets/f779797b-ec21-412e-baf3-0488d09e6bef)

But when I create the caption, it's still left-aligned. Is it anyway to make figure caption centered? Also, I understand that "Image Caption" style is used on both figure and table captions. Is there any way to bypass this so that ONLY figure captions are centralized?

```
---
format:
  docx:
    reference-doc: "templates/template.docx"
---

```{r}
#| label: fig-figure1
#| fig-cap: "Cars!"
plot(cars)
```
Interesting enough, in the output word file, if we change the caption to another type and then back to "Image Caption", we can see that the caption gets centralized:

![Presentation1-ezgif com-video-to-gif-converter (1)](https://github.com/user-attachments/assets/de064fc1-a287-45a5-b2fe-d1e16f8a9ded)

**Comment by mcanouil** [comment]
The issue is because the alignment does not come from the style but seems to be added/hardcoded on top of it.

Let's modify the Image Caption style in obvious ways in addition to make it centred.
<img width="620" alt="image" src="https://github.com/user-attachments/assets/ca33fcee-4570-4488-8e1e-a8c527c4da9d">

Generate the Word document

<table>
<tr><th>Input</th><th>Output</th></tr>
<tr><td>

````qmd
---
title: "Quarto Playground"
format:
  docx:
    reference-doc: centre-caption.docx
---

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

````

</td><td>

<img width="633" alt="image" src="https://github.com/user-attachments/assets/6f0b7c2b-f69d-41d0-a25e-f05845a3e8d3">

</td></tr>
</table>

If you look at the style of the caption, it get "+ Left"
<img width="1236" alt="Screenshot 2024-12-01 at 15 18 57" src="https://github.com/user-attachments/assets/378dc0c9-418a-4e57-92b5-44bec4bc9c0f">

Now, if you use standard figure (not cross-referenceable):

<table>
<tr><th>Input</th><th>Output</th></tr>
<tr><td>

````qmd
---
title: "Quarto Playground"
format:
  docx:
    reference-doc: centre-caption.docx
---

![An image]({{< placeholder 600 400 >}})

````

</td><td>

<img width="644" alt="image" src="https://github.com/user-attachments/assets/c221ab66-25b3-4994-aa99-6faa1395e75c">

</td></tr>
</table>

@cscheid Is it on purpose that the alignment is set this way for Word? This means you no longer can change caption alignment via reference document which is the only way for this format. (known behaviour in the feature matrix)
I believe for Word (and Powerpoint), it should only use the default style with nothing hardcoded unless `fig-align` is explicitly provided.

**Comment by mcanouil** [comment]
Alignment is hardcoded by Quarto

- https://github.com/quarto-dev/quarto-cli/blob/39baf4081f184fe090974295b48d1beab32cd1e4/src/resources/filters/layout/docx.lua#L97-L110

In theory, `fig-align` is supposed to control caption but it's not the case (maybe related to the cross-reference system refactoring/improvement)

<table>
<tr><th>Input</th><th>Output</th></tr>
<tr><td>

````qmd
---
title: "Quarto Playground"
format:
  docx:
    reference-doc: centre-caption.docx
    fig-align: center
---

![An image 1]({{< placeholder 600 400 >}}){#fig-placeholder-1}

![An image 2]({{< placeholder 600 400 >}}){#fig-placeholder-2 fig-align="right"}

````

</td><td>

<img width="969" alt="image" src="https://github.com/user-attachments/assets/d43af8b3-f728-4961-a19f-b4242f888614">

</td></tr>
</table>

Now, if I remove the alignment hardcoded/added, in addition to get `fig-align` working, you also get the reference document works by default as one might expect.

```diff
function docxParaStyles(align)
  local styles = "<w:pPr>\n"
- local captionAlign = docxAlign(align)
- if captionAlign then
-   styles = styles .. 
-       "<w:jc w:val=\"" .. captionAlign .. "\"/>\n"
- end  
  styles = styles ..
    "<w:spacing w:before=\"200\" />\n" ..
    "<w:pStyle w:val=\"ImageCaption\" />\n" ..
    "</w:pPr>\n"
  return styles
end
```

<table>
<tr><th>Input</th><th>Output</th></tr>
<tr><td>

````qmd
---
title: "Quarto Playground"
format:
  docx:
    reference-doc: centre-caption.docx
    fig-align: center
---

![An image 1]({{< placeholder 600 400 >}}){#fig-placeholder-1}

![An image 2]({{< placeholder 600 400 >}}){#fig-placeholder-2 fig-align="right"}

````

</td><td>

<img width="963" alt="image" src="https://github.com/user-attachments/assets/ce27ecc2-1867-4df1-996f-243c31c521ad">

</td></tr>
<tr><td>

````qmd
---
title: "Quarto Playground"
format:
  docx:
    reference-doc: centre-caption.docx
---

![An image 1]({{< placeholder 600 400 >}}){#fig-placeholder-1}

![An image 2]({{< placeholder 600 400 >}}){#fig-placeholder-2 fig-align="right"}

````

</td><td>

<img width="963" alt="image" src="https://github.com/user-attachments/assets/ce27ecc2-1867-4df1-996f-243c31c521ad">

</td></tr>
</table>

Unfortunately, the code which adds the alignment is needed for table caption to be aligned according to the document level `tbl-align`.

For short, the current implementation is in conflict with itself resulting in more bad results than good.

<details><summary>Quarto document</summary>
<p>

````qmd
---
title: "Quarto Playground"
format:
  docx:
    reference-doc: centre-caption.docx
    tbl-align: left
---

![An image 1]({{< placeholder 600 400 >}}){#fig-placeholder-1}

| A | B | C |
|---|---|---|
| 1 | 2 | 3 |
| 4 | 5 | 6 |

: A table 1 {#tbl-table-1}

![An image 2]({{< placeholder 600 400 >}}){#fig-placeholder-2 fig-align="right"}

| A | B | C |
|---|---|---|
| 1 | 2 | 3 |
| 4 | 5 | 6 |

: A table 2 {#tbl-table-2 tbl-align="right"}

````

</p>
</details> 

<table>
<tr><th>Version</th><th>Output</th></tr>
<tr><td>v1.7</td><td>

<img width="960" alt="image" src="https://github.com/user-attachments/assets/93f8fb99-cbc3-44b6-a975-c15edfd21bc9">

</td><td>

- "Figure 1" should be centred
- "Figure 2" should be right-aligned
- "Table 1" is correctly left-aligned
- "Table 2" should be right-aligned

Results: 1 correct out of 4 cases

</td></tr>
<tr><td>No alignment added in `docxDivCaption`</td><td>

<img width="959" alt="image" src="https://github.com/user-attachments/assets/3f5526a7-5c93-4adc-b2f7-45abc0b8ba93">

</td><td>

- "Figure 1" is correctly centred
- "Figure 2" is correctly right-aligned
- "Table 1" should be left-aligned
- "Table 2" is correctly right-aligned

Results: 3 correct out of 4 cases

</td></tr>
</table>

**Reply by mcanouil** [reply, marked as answer]
I've opened an issue as I do believe this to be a bug in the codebase that needs fixing.

- #11589

**Reply by Liu499-create** [reply]
I countered the same issue, but it remains unresolved. Is there a possibility it will be addressed in the next version?
